### ISSUE TITLE
Default values in totals row with GROUP BY WITH TOTALS

### ISSUE THREAD

**Issue by pic-Nick** [ClickHouse/ClickHouse#10433]
**Describe the unexpected behaviour**
If using GROUP BY WITH TOTALS on numeric field, the new row with totals has default value 0 (zero) in that field. In order to distinguish this new row from other row with value 0 (zero) as regular group key I add dummy field with constant string value. If I add this field in inner query, the totals row has the default value of empty string. But if I add this field in the same query where GROUP BY clause are, the totals row has my constant string value instead of empty string.

**How to reproduce**
Wrong result:
```
select k, 'R' gk, sum(v) s from (
  select 0 as k, 5 as v
  union all
  select 2, 5
  union all
  select 0, 3
  union all
  select 4, 9
)
group by k, gk with totals
```
Right result:
```
select k, gk, sum(v) s from (
  select 0 as k, 'R' gk, 5 as v
  union all
  select 2, 'R', 5
  union all
  select 0, 'R', 3
  union all
  select 4, 'R', 9
)
group by k, gk with totals
```

**Expected behavior**
Field in totals row should has empty string value in both cases.

**Comment by alexey-milovidov** [comment]
You can wrap your column in `toNullable` function:

```
select toNullable(k) AS grp, 'R' gk, sum(v) s from (
  select 0 as k, 5 as v
  union all
  select 2, 5
  union all
  select 0, 3
  union all
  select 4, 9
)
group by grp, gk with totals
```

```
┌─grp─┬─gk─┬─s─┐
│   0 │ R  │ 8 │
│   2 │ R  │ 5 │
│   4 │ R  │ 9 │
└─────┴────┴───┘

Totals:
┌──grp─┬─gk─┬──s─┐
│ ᴺᵁᴸᴸ │ R  │ 22 │
└──────┴────┴────┘
```

**Comment by pic-Nick** [comment]
Yes, this is suitable solution. This way I don't need dummy string column at all.

But... This column still has value 'R' in totals. Not default empty string. Is it right?

**Comment by alexey-milovidov** [comment]
This is not quite right. It's ok from the implementation point of view. We ensure that main result and totals have the same structure and "structure" means data types and values of constant expressions...
